new-docs/content/any/project/internals.md

```diff
@@ -40,3 +40,94 @@ library provides answers by inspecting application data, and control passes back
 and forth until the dialog terminates with a final “yes” or a “no” answer to the
 original authorization question. The virtual machine halts, and the library
 returns the answer back to your application as the authorization decision.
+
+
+## Data Filtering
+
+Oso supports applying authorization logic at the ORM layer so that you can
+efficiently authorize entire data sets. For example, suppose you have millions
+of posts in a social media application created by thousands of users, and
+regular users are only authorized to view posts from their friends. It would be
+inefficient to fetch all of the posts and authorize them one by one. It would
+be much more efficient to distill from the policy a _filter_ that can be
```

Comment:
maybe "wasteful and efficient" or "inefficient and better" instead of "inefficient and efficient"